pkg/proxy/util/conntrack.go

```diff
@@ -56,3 +57,34 @@ func ExecConntrackTool(execer exec.Interface, parameters ...string) error {
 	}
 	return nil
 }
+
+// ClearUDPConntrackForPort uses the conntrack tool to delete the conntrack entries
+// for the UDP connections specified by the port.
+// When a packet arrives, it will not go through NAT table again, because it is not "the first" packet.
+// The solution is clearing the conntrack. Known issues:
+// https://github.com/docker/docker/issues/8795
+// https://github.com/kubernetes/kubernetes/issues/31983
+func ClearUDPConntrackForPort(execer exec.Interface, port int) {
+	glog.V(2).Infof("Deleting conntrack entries for udp connections")
+	if port > 0 {
+		err := ExecConntrackTool(execer, "-D", "-p", "udp", "--dport", strconv.Itoa(port))
+		if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
+			glog.Errorf("conntrack return with error: %v", err)
+		}
+	} else {
+		glog.Errorf("Wrong port number. The port number must be greater than zero")
+	}
+}
+
+// DeleteServiceConnection uses the conntrack tool to delete the conntrack entries
+// for the UDP connections specified by the {origin, dest} IP pair.
+func DeleteUDPConnections(execer exec.Interface, origin, dest string) {
```

Comment:
> This should return the error, not log.

Yes, fixed.

> Should be named ClearUDPConntrackForPeers or something?

Yes, I agree `ClearUDPConntrackForPeers` should be a more descriptive name. Thanks!